tests/test_download.py

```diff
@@ -260,10 +260,13 @@ def test_https_connection(self):
     # 4: run with an HTTPS certificate that is expired
     # Be sure to offset from the port used in setUp to avoid collision.
 
-    port1 = self.server_process_handler.port + 1
-    port2 = self.server_process_handler.port + 2
-    port3 = self.server_process_handler.port + 3
-    port4 = self.server_process_handler.port + 4
+    # Use different ranges for each one of the ports to make sure
+    # that there are no collisions between the values which are
+    # generated for them.
+    port1 = random.randint(30000, 35000)
+    port2 = random.randint(35001, 40000)
+    port3 = random.randint(40001, 45000)
+    port4 = random.randint(45001, 50000)
```

Comment:
Hmmm, this now makes comment on line 261 untrue I think

Reply:
I am planning to add a function `_generate_port()` in the TestServerProcess class where I will add a function that generates a port and checks if the port is busy.
So, this code will be removed.